src/Domain/Bootstrap.php

```diff
@@ -155,13 +155,11 @@ protected function remove_core_blocks() {
 	 * Define the global feature flag
 	 */
 	protected function define_feature_flag() {
-		$woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
-		$flag        = 'stable';
-		if ( is_array( $woo_options ) && (
-				'experimental' === $woo_options['woocommerce_blocks_phase'] ||
-				'stable' === $woo_options['woocommerce_blocks_phase']
-		) ) {
-			$flag = sanitize_key( $woo_options['woocommerce_blocks_phase'] );
+		if ( getenv( 'WOOCOMMERCE_BLOCKS_PHASE' ) ) {
+			$flag = getenv( 'WOOCOMMERCE_BLOCKS_PHASE' );
```

Comment:
So the only problem here, is while `$woo_options` is sanitized (only allows `experimental` as the value), you're assigning the environment variable for the flag without any sanitization. I'd suggest something like:

```php
$allowed_flags = [ 'experimental', 'stable' ];
$flag = getenv( 'WOOCOMMERCE_BLOCKS_PHASE' );
if ( ! in_array( $flag, $allowed_flags ) ) {
    $woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
			$flag        = is_array( $woo_options ) && 'experimental' === $woo_options['woocommerce_blocks_phase'] ? 'experimental' : 'stable';
}
define( 'WOOCOMMERCE_BLOCKS_PHASE', $flag );
```

Reply:
Interesting, I assumed env must be safe but we can’t know for sure, I'm going to roll with your suggestion here.

Reply:
One could definitely argue that if someone has gained access to a system and is able to set environment variables, you have a bigger problem. I generally just like to be in the habit of sanitizing _any_ user space type data. We know exactly what data we want here, so let's make sure it's valid. Our future selves might thank us.